envoy/network/socket.h

```diff
@@ -228,6 +228,11 @@ class ConnectionInfoProvider {
    */
   virtual absl::string_view requestedServerName() const PURE;
 
+  /**
+   * @return requested application protocol(s) (e.g. ALPN in TLS), if any.
+   */
+  virtual const std::vector<std::string>& requestedApplicationProtocols() const PURE;
```

Comment:
Can you clarify why you need to add this? Is this to support filter chain matcher?

Reply:
not really, I think there is no way to let network filter to get the this info due to interface limit to let upstream also consume this info.